jellyfin/jellyfin-androidtv
# Add Playback speed controls

**Changes**
Adds Playback speed controls:
Use the same icon / general UX style as Jellyfin-Android's playback button
A playback speed will apply to the current and future videos for the current application launch. I.e. setting speed to 2.00x then playing the next episode will also play at 2.00x

This PR Contains the following:
- Add some unit tests for functionality I've touched where it's possible to automatically test
- Add a `VideoSpeedController` to keep logic / state nice and encapsulated with the associated `PlaybackSpeedAction`. This should hopefully make it easy to incorporate into #1057 

See each commit for more detailed descriptions.

This is my first foray into Kotlin, and I haven't done Android dev in 4-5 years. So I'm happy to take any/all criticisms, feedback and change ideas.

**Issues**
Fixes: #1189 

**Testing**
- Try playing a video with speed set to various levels
- Try seeking around in the video, the speed should be preserved
- Play a different video. It should continue at the same speed as the last video
- Ensure unit tests run / pass
